firebase-firestore/src/main/java/com/google/firebase/firestore/FirebaseFirestore.java

```diff
@@ -635,4 +636,15 @@ void validateReference(DocumentReference docRef) {
           "Provided document reference is from a different Cloud Firestore instance.");
     }
   }
+
+  /**
+   * Sets the language of the public API in the format of "gl-<language>/<version>" where version
+   * might be blank, e.g. `gl-cpp/`. The provided string is used as is.
+   *
+   * Note: this method is package-private because it is expected to only be called via JNI (which
+   * ignores access modifiers).
+   */
+  static void setClientLanguage(String languageToken) {
```

Comment:
Proguard will rewrite this method name in release builds of users' applications. We need to annotate this, likely with [`@KeepForSdk`](https://github.com/firebase/firebase-android-sdk#keepforsdk) or possibly `@Keep`, though I don't know which applies to package-private methods.

Note that this method does not have to be `public` to be accessible from JNI (JNI ignores access control modifiers), but this can only work when the name isn't shortened.

The argument here should have a nullability annotation, probably `@NonNull`.

Reply:
Hmm, looking at the way `@KeepForSdk` is described, it appears to have a more limited effect than `@Keep`, so it seems using `@Keep` is safer (apparently `@KeepForSdk` still allows for the method to be stripped in end-developer apps). Feel free to correct me, though.